src/Dependencies/Collections/SegmentedDictionary`2.cs

```diff
@@ -32,14 +32,28 @@ namespace Microsoft.CodeAnalysis.Collections
     internal sealed class SegmentedDictionary<TKey, TValue> : IDictionary<TKey, TValue>, IDictionary, IReadOnlyDictionary<TKey, TValue>
         where TKey : notnull
     {
+#if NETCOREAPP
```

Comment:
Nit: consider putting only the assignment in the ifdef, rather than the whole thing.